packages/qdrant-loader-core/src/qdrant_loader_core/graph/extractor/base_extractor.py

```
@@ -0,0 +1,225 @@
+)
+
+
+# ------------------------
+# EntityExtractor Interface
+# ------------------------
+class EntityExtractor(ABC):
+    """
+    Interface for all entity extractors.
+    """
+
+    _registry: dict[str, type[EntityExtractor]] = {}
```

Ruff (0.15.15)

[warning] Mutable default value for class attribute

(RUF012)